cachi2/core/package_managers/rpm/main.py

```diff
@@ -31,6 +33,127 @@
 READ_CHUNK = 1048576
 
 
+@dataclass
+class Package:
+    """An RPM package with relevant data for the SBOM generation."""
+
+    name: str
+    version: str
+    release: str
+    arch: str
+    download_url: str
+    epoch: Optional[str] = None
+    vendor: Optional[str] = None
+    checksum: Optional[str] = None
+    repository_id: Optional[str] = None
+
+    @classmethod
+    def from_filepath(cls, rpm_filepath: Path, rpm_download_metadata: dict[str, Any]) -> "Package":
+        """Instantiate a package dataclass instance from a download RPM file path."""
+        kwargs: dict[str, Optional[str]]
+
+        # we need this "hack" to silence mypy complaining about wrong dict types
+        _ = cls._query_rpm_fields(rpm_filepath)
+        kwargs = {**_}
```

Comment:
Yes, that's what we get for not using a statically typed language, sigh... I guess in this particular case mypy can't be blamed much though (a rare case) because we'd really assign a dict with immutable values to a dict that can host values of None. Any statically typed language would employ automatic type casting which is too much for mypy I guess, even though I could imagine mypy allowing assignments to a more relaxed type and just shut up.

Reply:
Just FYI, it's not so much a hack as just good defensive programming

```python
d_strings: dict[str, str] = cls._query_rpm_fields(rpm_filepath)
d_optstrings = d_strings

d_optstrings["foo"] = None  # and we just broke the type of d_strings
```

Perhaps a cleaner-looking solution would be

```python
kwargs: dict[str, Optional[str]] = {}
kwargs.update(cls._query_rpm_fields(rpm_filepath))
```

Reply:
>  Any statically typed language would employ automatic type casting which is too much for mypy I guess, even though I could imagine mypy allowing assignments to a more relaxed type and just shut up.

Now that I think about this, it's the same thing as `const char * -> char *` in C and GCC will complain about const correctness, so the relaxed type would actually be a very wrong thing to do.

> Perhaps a cleaner-looking solution would be
>
>kwargs: dict[str, Optional[str]] = {}
>kwargs.update(cls._query_rpm_fields(rpm_filepath))

Looks nicer, will adopt!